### DISCUSSION TITLE
Synthetise (fake) data with Azure Open AI

### DISCUSSION THREAD

**Discussion by rabbyn** [microsoft/presidio#1069] [Requested features] [answered]
Hello, I like the "synthetise" option to create fake data in the [demo app](https://huggingface.co/spaces/presidio/presidio_demo) , would it be possible to make a version compatible with Azure Open AI ?

https://github.com/microsoft/presidio/blob/0f48e6eb110c522f90a09db600482b62e97d113f/docs/samples/python/streamlit/openai_fake_data_generator.py#L3

**Comment by omri374** [comment, marked as answer]
Hi @rabbyn, the change is quite straightforward.

It requires to change the openai configuration on [this file](https://huggingface.co/spaces/presidio/presidio_demo/blob/main/openai_fake_data_generator.py) to something like this: https://github.com/MicrosoftLearning/mslearn-openai/blob/main/Instructions/Labs/02-natural-language-azure-openai.md

**Reply by omri374** [reply]
Hi @rabbyn, our new demo can be configured to use Azure OpenAI as well: https://huggingface.co/spaces/presidio/presidio_demo
You can set the env variable `OPENAI_TYPE` to `Azure`, or create an `.env` file containing `OPENAI_TYPE="Azure"`

**Reply by rabbyn** [reply]
Very good @omri374 , I will share that with my management. Thanks
